1-Authentication/0-sign-in-vanillajs/App/public/authRedirect.js

```diff
@@ -1,97 +1,100 @@
-    } else if (currentAccounts.length === 1) {
-        username = currentAccounts[0].username
-        welcomeUser(currentAccounts[0].username);
-        updateTable(currentAccounts[0]);
-    }
-}
-
-function handleResponse(response) {
-
-    /**
-     * To see the full list of response object properties, visit:
-     * https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/dev/lib/msal-browser/docs/request-response-object.md#response
-     */
-
-    if (response !== null) {
-        username = response.account.username
-        welcomeUser(username);
-        updateTable(response.account);
-    } else {
-        selectAccount();
-
-        /**
-         * If you already have a session that exists with the authentication server, you can use the ssoSilent() API
-         * to make request for tokens without interaction, by providing a "login_hint" property. To try this, comment the 
-         * line above and uncomment the section below.
-         */
-
-        // myMSALObj.ssoSilent(silentRequest).
-        //     then((response) => {
-        //         welcomeUser(response.account.username);
-        //         updateTable(response.account);
-        //     }).catch(error => {
-        //         console.error("Silent Error: " + error);
-        //         if (error instanceof msal.InteractionRequiredAuthError) {
-        //             signIn();
-        //         }
-        //     });
-    }
-}
-
-function signIn() {
-
-    /**
-     * You can pass a custom request object below. This will override the initial configuration. For more information, visit:
-     * https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/dev/lib/msal-browser/docs/request-response-object.md#request
-     */
-
-    myMSALObj.loginRedirect(loginRequest);
-}
-
-function signOut() {
-
-    /**
-     * You can pass a custom request object below. This will override the initial configuration. For more information, visit:
-     * https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/dev/lib/msal-browser/docs/request-response-object.md#request
-     */
-
-    // Choose which account to logout from by passing a username.
-    const logoutRequest = {
-        account: myMSALObj.getAccountByUsername(username),
-        postLogoutRedirectUri: '/signout', // remove this line if you would like navigate to index page after logout.
-
-    };
-
-    myMSALObj.logoutRedirect(logoutRequest);
-}
+// Create the main myMSALObj instance
+// configuration parameters are located at authConfig.js
+const myMSALObj = new msal.PublicClientApplication(msalConfig);
+
+myMSALObj.initialize().then(() => {
+/**
+ * A promise handler needs to be registered for handling the
+ * response returned from redirect flow. For more information, visit:
+ * https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/dev/lib/msal-browser/docs/initialization.md#redirect-apis
+ */
+myMSALObj.handleRedirectPromise()

```

Comment:
Wrapped this in a promise to make sure it doesn't run before the myMSALObj is initialized.